### PR TITLE
Fix and improve handling of input upload problems while running tool tests.

### PR DESCRIPTION
**Pull request by jmchilton**
Previously these would just cause bad tests to pass without any output.

### CODE REVIEW

File: lib/galaxy/tool_util/verify/interactor.py [hunk 1]
```diff
@@ -286,10 +286,14 @@ def test_data_path(self, tool_id, filename):
...
 
     @nottest
-    def test_data_download(self, tool_id, filename, mode='file'):
+    def test_data_download(self, tool_id, filename, mode='file', is_output=True):
         if self.supports_test_data_download:
             response = self._get(f"tools/{tool_id}/test_data_download?filename={filename}", admin=True)
-            assert response.status_code == 200, "Test file (%s) is missing. If you use planemo try --update_test_data to generate one." % filename
+            if response.status_code != 200:
+                if is_output:
+                    raise AssertionError(f"Test output file ({filename}) is missing. If you use planemo try --update_test_data to generate one.")
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/verify/interactor.py]:
```suggestion
                    raise AssertionError(f"Test output file ({filename}) is missing. If you are using planemo, try adding --update_test_data to generate it.")
```

File: lib/galaxy/tool_util/verify/interactor.py [hunk 2]
```diff
@@ -952,15 +956,20 @@ def verify_tool(tool_id,
...
+        try:
+            stage_data_in_history(galaxy_interactor,
+                                tool_id,
+                                testdef.test_data(),
+                                history=test_history,
+                                force_path_paste=force_path_paste,
+                                maxseconds=maxseconds)
+        except Exception as e:
+            input_staging_exception = e
+            raise e
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/verify/interactor.py]:
```suggestion
            raise
```

File: lib/galaxy/tool_util/verify/interactor.py [hunk 3]
```diff
@@ -952,15 +956,20 @@ def verify_tool(tool_id,
...
-                            history=test_history,
-                            force_path_paste=force_path_paste,
-                            maxseconds=maxseconds)
+        try:
+            stage_data_in_history(galaxy_interactor,
+                                tool_id,
+                                testdef.test_data(),
+                                history=test_history,
+                                force_path_paste=force_path_paste,
+                                maxseconds=maxseconds)
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/verify/interactor.py]:
```suggestion
            stage_data_in_history(
                galaxy_interactor,
                tool_id,
                testdef.test_data(),
                history=test_history,
                force_path_paste=force_path_paste,
                maxseconds=maxseconds,
            )
```

**Comment by mvdbeek** [comment]
Very cool!

**Pull request merged**